packages/wrangler/src/config/validation.ts

```diff
@@ -461,6 +461,84 @@ function normalizeAndValidateModulePaths(
   return mapping;
 }
 
+/**
+ * Check whether a value has the shape of a route, which can be a string
+ * or an object that looks like {pattern: string, zone_id: string }
+ */
+function isValidRouteValue(item: unknown): boolean {
+  return (
+    !!item &&
+    (typeof item === "string" ||
+      (typeof item === "object" &&
+        // @ts-expect-error how do i refine this type?
```

Comment:
🤷‍♂️🤷‍♂️🤷‍♂️ I dunno what to do here. I tried `"pattern" in item` but it didn't work like I thought it would